mlrun/datastore/sources.py

```diff
@@ -412,6 +413,51 @@ def to_dataframe(
             **reader_args,
         )
 
+    def _get_spark_additional_filters(self):
+        if not self.additional_filters:
+            return None
+        none_values = [None, np.nan, np.NaN]
+        additional_filters = deepcopy(self.additional_filters)
+        is_first = True
+        spark_filter = ""
+        for filter_tuple in additional_filters:
+            if not filter_tuple:
+                continue
+            if not is_first:
+                spark_filter += " AND "
+            else:
+                is_first = False
+            col_name, op, value = filter_tuple
+            if isinstance(value, (str, datetime)):
+                value = rf"'{value}'"
+            if isinstance(value, (list, tuple, set, np.ndarray)) and (
```

Comment:
Why `np.ndarray`? According to [pyarrow docs](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetDataset.html), it must be
> list, a set or a tuple

Reply:
Nice